src/System.IO.Pipes.AccessControl/tests/AnonymousPipeTests/AnonymousPipeServerStreamAclTests.cs

```diff
@@ -0,0 +1,212 @@
+        [InlineData(PipeAccessRights.FullControl,             AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.Read,                    AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.Read,                    AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.ReadAttributes,          AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.ReadAttributes,          AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.ReadData,                AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.ReadData,                AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.ReadExtendedAttributes,  AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.ReadExtendedAttributes,  AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.ReadPermissions,         AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.ReadPermissions,         AccessControlType.Deny)]
+        // [InlineData(PipeAccessRights.ReadWrite, AccessControlType.Allow)] Combination tested in Create_ValidSecurity
+        [InlineData(PipeAccessRights.ReadWrite,               AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.Synchronize,             AccessControlType.Allow)]
+        // [InlineData(PipeAccessRights.Synchronize, AccessControlType.Deny)] Combination tested in Create_InvalidMaskSecurityConfiguration
+        [InlineData(PipeAccessRights.TakeOwnership,           AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.TakeOwnership,           AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.Write,                   AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.Write,                   AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.WriteAttributes,         AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.WriteAttributes,         AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.WriteData,               AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.WriteData,               AccessControlType.Deny)]
+        [InlineData(PipeAccessRights.WriteExtendedAttributes, AccessControlType.Allow)]
+        [InlineData(PipeAccessRights.WriteExtendedAttributes, AccessControlType.Deny)]
+        // [InlineData(PipeAccessRights.ReadWrite | PipeAccessRights.Synchronize, AccessControlType.Allow)] Combination tested in Create_ValidSecurity
+        [InlineData(PipeAccessRights.ReadWrite | PipeAccessRights.Synchronize, AccessControlType.Deny)]
+        public void Create_UnauthorizedSecurityConfiguration(PipeAccessRights rights, AccessControlType accessControl)
+        {
+            PipeSecurity security = GetPipeSecurity(WellKnownSidType.BuiltinUsersSid, rights, accessControl);
+
+            Assert.Throws<UnauthorizedAccessException>(() =>
+            {
+                CreateAndVerifyAnonymousPipe(PipeDirection.In, HandleInheritability.None, 0, security).Dispose();
```

Comment:
You might want to break this into two separate helpers. What is throwing? Setting the rights or trying to read what they are? You can't tell from this, and I suspect that creating the pipe doesn't always throw `UnauthorizedAccessException`.

Reply:
I change this to address your suggestion. Let me know if you agree, @JeremyKuhne .